compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerationExtension.kt

```diff
@@ -75,7 +79,9 @@ internal class CodeGenerationExtension(
     val generatedFiles = mutableMapOf<String, GeneratedFile>()
 
     val (privateCodeGenerators, nonPrivateCodeGenerators) =
-      codeGenerators.partition { it is PrivateCodeGenerator }
+      codeGenerators
+        .filter { it.isApplicable(anvilContext) }
```

Comment:
I had to move the check here in order to get the context. It also feels cleaner. It shouldn't be the responsibility of the registrar.